### PR TITLE
Adding Identity to ShapeMenuItem

### PR DESCRIPTION
**Pull request by thekaveman**
This is a fix for #7789

### CODE REVIEW

**Comment by thekaveman** [comment]
I see that this was already added in the `dev` branch from #7040 (there wasn't an issue so I missed it before submitting mine), but I feel like it should be in the `1.10.x` branch, no?

File: src/Orchard.Web/Core/Navigation/Migrations.cs [hunk 1]
```diff
@@ -177,5 +181,13 @@ public class Migrations : DataMigrationImpl {
 
             return 6;
```
Comment (sebastienros) [inline src/Orchard.Web/Core/Navigation/Migrations.cs]: `return 7;`

Comment (thekaveman) [inline src/Orchard.Web/Core/Navigation/Migrations.cs]: Isn't that the `UpdateFrom5` method? 

`return 7;` is in `UpdateFrom6` below.

Comment (sebastienros) [inline src/Orchard.Web/Core/Navigation/Migrations.cs]: Right, the diff confused me. But then move all new WithIdentity you added to the `UpdateFrom6` as new installs will always go there.

File: src/Orchard.Web/Core/Navigation/Migrations.cs [hunk 2]
```diff
@@ -30,6 +30,7 @@ public class Migrations : DataMigrationImpl {
             ContentDefinitionManager.AlterTypeDefinition("Menu", cfg => cfg
                 .WithPart("CommonPart", p => p.WithSetting("OwnerEditorSettings.ShowOwnerEditor", "false"))
                 .WithPart("TitlePart")
+                .WithIdentity()
```
Comment (sebastienros) [inline src/Orchard.Web/Core/Navigation/Migrations.cs]: This should only be in `UpdateFrom6`

File: src/Orchard.Web/Core/Navigation/Migrations.cs [hunk 3]
```diff
@@ -68,6 +69,7 @@ public class Migrations : DataMigrationImpl {
                     .WithPart("ShapeMenuItemPart")
                     .WithPart("MenuPart")
                     .WithPart("CommonPart")
+                    .WithIdentity()
```
Comment (sebastienros) [inline src/Orchard.Web/Core/Navigation/Migrations.cs]: This should only be in `UpdateFrom6`

File: src/Orchard.Web/Core/Navigation/Migrations.cs [hunk 4]
```diff
@@ -101,6 +103,7 @@ public class Migrations : DataMigrationImpl {
             ContentDefinitionManager.AlterTypeDefinition("Menu", cfg => cfg
                 .WithPart("CommonPart")
                 .WithPart("TitlePart")
+                .WithIdentity()
```
Comment (sebastienros) [inline src/Orchard.Web/Core/Navigation/Migrations.cs]: This should only be in `UpdateFrom6`

File: src/Orchard.Web/Core/Navigation/Migrations.cs [hunk 5]
```diff
@@ -151,6 +154,7 @@ public class Migrations : DataMigrationImpl {
                     .WithPart("ShapeMenuItemPart")
                     .WithPart("MenuPart")
                     .WithPart("CommonPart")
+                    .WithIdentity()
```
Comment (sebastienros) [inline src/Orchard.Web/Core/Navigation/Migrations.cs]: This should only be in `UpdateFrom6`

**Comment by thekaveman** [comment]
Thanks for the feedback. Since `UpdateFrom5` already adds identity to `Menu`, I've just left that one out completely.

**Pull request merged**